src/attributes.h

```diff
@@ -16,4 +16,12 @@
 #  define LIFETIMEBOUND
 #endif
 
+#if defined(__GNUC__)
+#  define ALWAYS_INLINE inline __attribute__((always_inline))
+#elif defined(_MSC_VER)
+#  define ALWAYS_INLINE __forceinline
+#else
+#  define ALWAYS_INLINE inline
```

Comment:
I think it'd be better to error in this case so that we can add another platform exception if necessary. IIRC for the places that we use it, inlining is critical and we might want to design functions differently for platforms where it's not possible.

Especially given the [gcc semantics](https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html):

> Failure to inline such a function is diagnosed as an error.

As-is, I think this is currently a theoretical regression as we might now compile on a platform that would be an error on master.

Reply:
Mind suggesting an error wording?

Reply:
How about: `#error No known always_inline attribute for this platform.`

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/27575#issuecomment-1535337768).